lib/io.toit

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2023 Toitware ApS. All rights reserved.
+
+/**
+A producer of bytes.
+
+The most important implementations of this interface are
+  $ByteArray and $string. However, any data structure that can be
```

Comment:
A lot of primitives only accept those. Internally, they are called blobs. Maybe they need a name that users can relate to? Not sure if they also need a type, but probably not.